### PR TITLE
08 post a comment

### PR DESCRIPTION
**Pull request by Pippa101**
created optimistic rendering to show user their comment has been added

### CODE REVIEW

**Review by pippaamy — CHANGES_REQUESTED** [review]
If you post more than one comment the page doesn't refresh and the comment does go through, the same with if you are offline and get the error message - once you're online the comment doesn't refresh. Once you post that comment if there’s more comments posted at once you could have your comment buried. Consider an alternative way to do this.

File: src/components/Comments.js [hunk 1]
```diff
@@ -27,15 +26,18 @@ const Comments = ({ article_id }) => {
...
       })
-      .then(() => {
-        setReloadComments(true);
+      .then((response) => {
+        setArticleComments((currArticles) => {
+          return [response, ...currArticles];
+        });
       })
       .catch((error) =>
         alert(
```
Comment (JVal-NC) [inline src/components/Comments.js]: I wouldn't recommend using alert as I think this is a pop up which isn't very UI friendly. Could we conditionally render a component instead?

File: src/components/Comments.js [hunk 2]
```diff
@@ -6,15 +6,14 @@ const Comments = ({ article_id }) => {
   const [isLoading, setIsLoading] = useState(true);
```
Comment (JVal-NC) [inline src/components/Comments.js:6]: On this component there's a couple bugs I think?

You can post a blank comment.
You post the same comment as the previous in the event you press the post button after the text area is cleared.
I think you may be able to press the button mid-post request also?

I think once these are addressed some UI feedback to inform the user what is going wrong would be great. Possibly a component saying 'post successful' or similar too?

**Review by JVal-NC — COMMENTED** [review]
Hi Pippa, good work. Some comments to include in your next PR attached.

**Review by pippaamy — COMMENTED** [review]
Hiya, can we have some a tag of some kind to tell us our comment hasn't been posted when offline? :)

**Review by pippaamy — APPROVED** [review]
Great work, go ahead and merge!

**Pull request merged**